### PR TITLE
Text area blog post

### PR DESCRIPTION
**Pull request by darrenburns**

### CODE REVIEW

File: docs/blog/posts/text-area-learnings.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+---
+
+# Things I learned while building Textual's TextArea
+
+`TextArea` is the latest widget to be added to Textual's [growing collection](https://textual.textualize.io/widget_gallery/).
+It provides a multi-line space to edit text, and features optional syntax highlighting for a selection of languages.
+
+![text-area-welcome.gif](../images/text-area-learnings/text-area-welcome.gif)
+
+Adding a `TextArea` to your Textual app is as simple adding this to your `compose` method:
```
Comment (davep) [inline docs/blog/posts/text-area-learnings.md:17]:
```suggestion
Adding a `TextArea` to your Textual app is as simple as adding this to your `compose` method:
```

File: docs/blog/posts/text-area-learnings.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+Thankfully it resulted in some wonderful Tetris-esque whiteboards along the way!
+
+<figure markdown>
+  ![cursor_position_updating_via_api.png](../images/text-area-learnings/cursor_position_updating_via_api.png){ loading=lazy }
+  <figcaption>A TetrisArea white-boarding session.</figcaption>
+</figure>
+
+Sometimes stepping away from the screen and scribbling on a whiteboard with your colleagues (thanks [Dave](https://fosstodon.org/@davep)!) is what's needed to finally crack a tough problem.
+
+Many thanks to [David Brochart](https://fosstodon.org/@davidbrochart@mastodon.top) for sending me down this rabbit hole!
```
Comment (davep) [inline docs/blog/posts/text-area-learnings.md:77]:
```suggestion
Many thanks to [David Brochart](https://mastodon.top/@davidbrochart) for sending me down this rabbit hole!
```

File: docs/blog/posts/text-area-learnings.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+## Spending a few minutes running a profiler can be really beneficial
+
+While building the `TextArea` widget I avoided heavy optimisation work that may have affected
```
Comment (davep) [inline docs/blog/posts/text-area-learnings.md:81]: I sense https://xkcd.com/1691/ belongs here. :-P

File: docs/blog/posts/text-area-learnings.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+Switching to `tuple` resulted in another noticeable increase in responsiveness.
+Key-press handling time dropped by almost 50%!
+Unfortunately, this change _does_ impact readability.
+However, the scope in which these tuples were used was very small, and so I felt it was a worthy trade-off.
+
+
+## Syntax highlighting is very different from what I expected
+
+In order support syntax highlighting, we make use of
```
Comment (davep) [inline docs/blog/posts/text-area-learnings.md:146]:
```suggestion
In order to support syntax highlighting, we make use of
```

**Review by davep — APPROVED** [review]
Not one mention of Lisp though. :-/

**Comment by TomJGooding** [comment]
> Not one mention of Lisp though. :-/

Perhaps a note about Lisp could be added in parentheses (or several hundreds).

File: docs/blog/posts/text-area-learnings.md (hunk 3 above)
Comment (darrenburns) [reply docs/blog/posts/text-area-learnings.md:81]: True 😂

**Review by willmcgugan — APPROVED** [review]
Awesome

**Pull request merged**